rust/src/run/aggregation.rs

```diff
@@ -0,0 +1,136 @@
+use std::{
+    fs::{create_dir_all, File},
+    io::Write,
+    path::Path,
+};
+
+use axiom_eth::{
+    halo2_base::{gates::circuit::CircuitBuilderStage, utils::fs::gen_srs},
+    halo2_proofs::{
+        dev::MockProver,
+        plonk::{keygen_pk, keygen_vk, ProvingKey, VerifyingKey},
+        SerdeFormat,
+    },
+    halo2curves::bn256::{Fr, G1Affine},
+    snark_verifier_sdk::{
+        halo2::aggregation::{AggregationCircuit, VerifierUniversality},
+        Snark, SHPLONK,
+    },
+    utils::{keccak::decorator::RlcKeccakCircuitParams, snark_verifier::AggregationCircuitParams},
+};
+use axiom_eth::{
+    rlc::circuit::RlcCircuitParams,
+    snark_verifier_sdk::{halo2::gen_snark_shplonk, CircuitExt},
+};
+use ethers::providers::{JsonRpcClient, Provider};
+
+use crate::{
+    scaffold::{AxiomCircuit, AxiomCircuitScaffold},
+    types::{AxiomCircuitParams, AxiomV2CircuitOutput, AxiomV2DataAndResults},
+    utils::build_axiom_v2_compute_query,
+};
+
+pub fn create_aggregation_circuit(
```

Comment:
better to add `stage` so we can create circuit with `CircuitBuilderStage::{Keygen,Prover}` for optimization.